ThunderCloud/StormLanguageController.swift

```diff
@@ -38,17 +45,33 @@ public class StormLanguageController: NSObject {
             return Locale(identifier: languageString)
         })
         
-        //Add override locales if they exist
-        if let overrideObject = UserDefaults.standard.object(forKey: "TSCLanguageOverride") as? Data, let _overrideLanguage = NSKeyedUnarchiver.unarchiveObject(with: overrideObject) as? TSCLanguage, let localeIdentifier = _overrideLanguage.languageIdentifier {
-            _preferredLocales.insert(Locale(identifier: localeIdentifier), at: 0)
-            overrideLanguage = _overrideLanguage
+        // If the user has applied an override language to the app we need to retrieve a saved version and apply it as the app language
+        // Check the defaults for the override language filename
+        if let overridePackFileName = UserDefaults.standard.object(forKey: overrideLanguagePackSavingKey) as? String {
+            
+            // If we have the saved override filename, filter our available language packs for it
+            let savedOverridePack = availableLanguagePacks?.filter({ (pack) -> Bool in
```

Comment:
You can just do a `first(where: {})` here :)

Reply:
👌